src/uproot/_dask.py

```diff
@@ -647,11 +649,14 @@ def real_filter_branch(branch):
         entry_start, entry_stop = _regularize_entries_start_stop(
             ttree.num_entries, None, None
         )
-        entry_step = 0
-        if uproot._util.isint(step_size):
-            entry_step = step_size
-        else:
-            entry_step = ttree.num_entries_for(step_size, expressions=common_keys)
+
+        branchid_interpretation = {}
+        for key in common_keys:
+            branch = ttree[key]
+            branchid_interpretation[branch.cache_key] = branch.interpretation
+        entry_step = _regularize_step_size(
+            ttree, step_size, entry_start, entry_stop, branchid_interpretation
+        )
```

Comment:
Quick question, @jpivarski

The above code is called for each ttree, and I've noticed that different trees with the same branch name and type often have different step-sizes. Why does step-size depend on the ttree even when the branch name and data type are the same?

Just curious.

Reply:
Oh right, understood. Thanks